src/ngResource/resource.js

```diff
@@ -431,16 +431,20 @@ angular.module('ngResource', ['ng']).
             }
             if (!(new RegExp("^\\d+$").test(param)) && param &&
               (new RegExp("(^|[^\\\\]):" + param + "(\\W|$)").test(url))) {
-              urlParams[param] = true;
+              urlParams[param] = { isQueryParam: (new RegExp("\\?(.*)\\:" + param)).test(url) };
```

Comment:
You need the `'(\\W|$)'` part or else you might get false positive. (Better yet `'(?:\\W|$)'`.)
E.g. `param: 'foo'` and `url: '/path/foo?key=:foobar'` will lead to a false positive.